includes/REST_API/Stories_Base_Controller.php

```diff
@@ -58,8 +68,155 @@ class Stories_Base_Controller extends WP_REST_Posts_Controller {
+					],
+				],
+				[
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => [ $this, 'get_lock' ],
+					'permission_callback' => [ $this, 'update_item_permissions_check' ],
+					'args'                => $lock_args,
+				],
+				[
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => [ $this, 'update_lock' ],
+					'permission_callback' => [ $this, 'update_item_permissions_check' ],
+					'args'                => $lock_args,
+				],
+				[
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => [ $this, 'delete_lock' ],
+					'permission_callback' => [ $this, 'update_item_permissions_check' ],
+					'args'                => $lock_args,
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get post lock
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success.
+	 */
+	public function get_lock( $request ) {
+		return $this->prepare_lock_for_response( $request );
+	}
+
+	/**
+	 * Update post lock
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success.
+	 */
+	public function update_lock( $request ) {
+		if ( ! function_exists( 'wp_set_post_lock' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/post.php';
+		}
+
+		wp_set_post_lock( $request['id'] );
+
+		return $this->prepare_lock_for_response( $request );
+	}
+
+	/**
+	 * Delete post lock
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response Response object on success.
+	 */
+	public function delete_lock( $request ) {
+		$previous = $this->prepare_lock_for_response( $request );
+		$result   = delete_post_meta( $request['id'], '_edit_lock' );
+		$data     = [];
+		if ( ! is_wp_error( $previous ) ) {
+			$data = $previous->get_data();
+		}
+		$response = new WP_REST_Response();
+		$response->set_data(
+			[
+				'deleted'  => $result,
+				'previous' => $data,
+			]
+		);
+		update_option( 'wibble', time() );
+
+		return $response;
+	}
+
+	/**
+	 * Prepares a single lock output for response.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object.
+	 */
+	public function prepare_lock_for_response( $request ) {
+		$lock = get_post_meta( $request['id'], '_edit_lock', true );
+
+		$nonce = wp_create_nonce( 'wp_rest' );
+		$data  = [
+			'locked' => false,
+			'nonce'  => $nonce,
+		];
+		$links = [];
+
+		if ( $lock ) {
+			$lock                 = explode( ':', $lock );
+			list ( $time, $user ) = $lock;
+
+			/** This filter is documented in wp-admin/includes/ajax-actions.php */
+			$time_window = apply_filters( 'wp_check_post_lock_window', 150 );
+
+			if ( $time && $time > time() - $time_window ) {
+				$data = [
+					'locked' => true,
+					'time'   => $time,
+					'user'   => (int) $user,
+					'nonce'  => $nonce,
```

Comment:
This is a new addition to make `navigator.sendBeacon` work? Could we pass it in `get_editor_settings()` instead?

Reply:
Nonces are time related and single use. We update this nonce everytime lock is refreshed, means that nonce should be valid when the delete command is sent.